apps/converter/htmlToMarkdown.js

```diff
@@ -0,0 +1,172 @@
+import crypto from 'crypto';
+import fs from 'fs-extra';
+import imageType from 'image-type';
+import { JSDOM } from 'jsdom';
+import fetch from 'node-fetch';
+import path from 'path';
+import TurndownService from 'turndown';
+import { fileURLToPath } from 'url';
+
+// @todo Fix this to work locally and live
+const isLagoon = !!process.env.LAGOON;
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = isLagoon
+  ? '/app/web/sites/default/files/converted'
+  : path.dirname(__filename);
+
+async function extractMainContent(htmlString) {
+  const bodyRegex = /<body[^>]*>([\s\S]*?)<\/body>/i;
+  const match = htmlString.match(bodyRegex);
+  // Return the captured group (content between tags) or null if no match
+  const html = match ? match[1] : null;
+
+  if (html) {
+    // Create a new JSDOM instance and parse the HTML string
+    const dom = new JSDOM(html);
+    // Extract the <main> element content
+    let mainElement = dom.window.document.querySelector('main');
+    if (!mainElement) {
+      mainElement = dom.window.document.querySelector('article');
+    }
```

Comment:
```suggestion
    const mainElement = dom.window.document.querySelector('main, article');
```

If `<main>` element does not exist, then it would find `<article>`

But not sure what occurs if both exist or multiple `article` exist 🤔

Reply:
@dan2k3k4 all these are correct, this is a WIP, just want to deliver something for the upcoming webinar (with controlled input).